Source/MQTTnet/Formatter/MqttPublishPacketFactory.cs

```diff
@@ -17,6 +17,8 @@ public MqttPublishPacket Clone(MqttPublishPacket publishPacket)
             {
                 Topic = publishPacket.Topic,
                 Payload = publishPacket.Payload,
+                PayloadOffset = publishPacket.PayloadOffset,
+                PayloadLength = publishPacket.PayloadLength,
```

Comment:
We have a problem when the value is not null here because several interceptors in the server allow for changing the payload. If the length is present in these cases existing code will fail because now the users have to update the length property accordingly.

Reply:
So, what should we do here? Or we can design two fields, `byte[] Payload` and `ArraySegment<byte> PayloadSegment`. When set the value of one field, it affects the obtaining of the other